actor/v7action/service_instance.go

```diff
@@ -50,6 +51,49 @@ func (actor Actor) UpdateUserProvidedServiceInstance(serviceInstanceName, spaceG
 	return Warnings(warnings), nil
 }
 
+func (actor Actor) UpdateManagedServiceInstance(serviceInstanceName, spaceGUID string, serviceInstanceUpdates resources.ServiceInstance) (Warnings, error) {
+	var (
+		original resources.ServiceInstance
+		jobURL   ccv3.JobURL
+	)
+
+	warnings, err := railway.Sequentially(
+		func() (warnings ccv3.Warnings, err error) {
+			original, _, warnings, err = actor.CloudControllerClient.GetServiceInstanceByNameAndSpace(serviceInstanceName, spaceGUID)
+			return
+		},
+		func() (warnings ccv3.Warnings, err error) {
+			if original.Type != resources.ManagedServiceInstance {
```

Comment:
Is there a reason for this check to be in its own function rather than as part of the previous one?

Reply:
It could be merged into the previous step, but I think having it here helps with separation of concerns. Also since you’d only want to do this if the previous step passed, it helps with consistency of error handling.